internal/db/storage/blob/s3/writer/writer.go

```diff
@@ -51,7 +52,9 @@ type Writer interface {
 func New(opts ...Option) Writer {
 	w := new(writer)
 	for _, opt := range append(defaultOpts, opts...) {
-		opt(w)
+		if err := opt(w); err != nil {
+			log.Warn(errors.ErrOptionFailed(err, reflect.ValueOf(opt)))
```

Comment:
![badge](https://img.shields.io/badge/review-imo-orange.svg?style=flat-square)
it is not enough information is provided in the log. there should be package name or something to know where the error is occurred.

Reply:
we can get the function name without passing the `opt`, but we need to do the code changes.
https://play.golang.org/p/LFL9F2Ix6fA

what do you think? @rinx @kpango

Reply:
the current output of the log:
```
2020-09-03 16:18:27	[WARN]:	failed to setup option :github.com/vdaas/vald/internal/db/storage/blob/s3/writer.WithErrGroup.func1: invalid option. name: errgroup, val: nil
```

Reply:
↑
I think the current implementation has enough information.
when we read this output, I think we notice the place of FOP :thinking:

@rinx

Reply:
I didn't know the detail of the implementation of ErrOptionFailed. okay.

Reply:
I didn't know that too 😂😂😂

Reply:
Sorry for my spiritual magic code lol